pyicloud/base.py

```diff
@@ -112,7 +112,10 @@ def request(self, method, url, **kwargs):  # pylint: disable=arguments-differ
                     # Handle re-authentication for Find My iPhone
                     LOGGER.debug("Re-authenticating Find My iPhone service")
                     try:
-                        self.service.authenticate(True, "find")
+                        # If 450, authentication requires a full sign in to the account
+                        service = None if response.status_code == 450 else 'find'
```

Comment:
It's always status code 450 here because that's checked in the above if-statement

Reply:
The check for a response.code 450 should actually include a check for the other response codes also requiring a reauthentication (421 & 500), not just a 450. Updated PR submitted.